amd_openvx_extensions/amd_rpp/source/tensor/Transpose.cpp

```diff
@@ -0,0 +1,296 @@
+    if (data->deviceType == AGO_TARGET_AFFINITY_GPU) {
+#if ENABLE_HIP
+        hipError_t err = hipHostMalloc(&data->pSrcGenericDesc, sizeof(RpptGenericDesc), hipHostMallocDefault);
+        if (err != hipSuccess)
+            return ERRMSG(VX_ERROR_NOT_ALLOCATED, "refresh: hipHostMalloc of size %ld failed \n", sizeof(RpptGenericDesc));
+        err = hipHostMalloc(&data->pDstGenericDesc, sizeof(RpptGenericDesc), hipHostMallocDefault);
+        if (err != hipSuccess)
+            return ERRMSG(VX_ERROR_NOT_ALLOCATED, "refresh: hipHostMalloc of size %ld failed \n", sizeof(RpptGenericDesc));
+#endif
+    } else if (data->deviceType == AGO_TARGET_AFFINITY_CPU) {
+        data->pSrcGenericDesc = new RpptGenericDesc;
+        data->pDstGenericDesc = new RpptGenericDesc;
+    }
+
+    // Querying for input tensor
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_NUMBER_OF_DIMS, &data->pSrcGenericDesc->numDims, sizeof(data->pSrcGenericDesc->numDims)));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DIMS, &data->inputTensorDims, sizeof(vx_size) * data->pSrcGenericDesc->numDims));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DATA_TYPE, &input_tensor_dtype, sizeof(input_tensor_dtype)));
+    data->pSrcGenericDesc->dataType = getRpptDataType(input_tensor_dtype);
+    data->pSrcGenericDesc->offsetInBytes = 0;
+    fillGenericDescriptionPtrfromDims(data->pSrcGenericDesc, data->inputLayout, data->inputTensorDims);
+
+    // Querying for output tensor
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_NUMBER_OF_DIMS, &data->pDstGenericDesc->numDims, sizeof(data->pDstGenericDesc->numDims)));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_DIMS, &data->outputTensorDims, sizeof(vx_size) * data->pDstGenericDesc->numDims));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_DATA_TYPE, &output_tensor_dtype, sizeof(output_tensor_dtype)));
+    data->pDstGenericDesc->dataType = getRpptDataType(output_tensor_dtype);
+    data->pDstGenericDesc->offsetInBytes = 0;
+    fillGenericDescriptionPtrfromDims(data->pDstGenericDesc, data->outputLayout, data->outputTensorDims); 
+    
+    refreshTranspose(node, parameters, num, data);
+    STATUS_ERROR_CHECK(createRPPHandle(node, &data->handle, data->inputTensorDims[0], data->deviceType));
+    STATUS_ERROR_CHECK(vxSetNodeAttribute(node, VX_NODE_LOCAL_DATA_PTR, &data, sizeof(data)));
+    return VX_SUCCESS;
+}
+
+static vx_status VX_CALLBACK uninitializeTranspose(vx_node node, const vx_reference *parameters, vx_uint32 num) {
+    TransposeLocalData *data;
+    STATUS_ERROR_CHECK(vxQueryNode(node, VX_NODE_LOCAL_DATA_PTR, &data, sizeof(data)));
+    STATUS_ERROR_CHECK(releaseRPPHandle(node, data->handle, data->deviceType));
+    if (data->deviceType == AGO_TARGET_AFFINITY_GPU) {
+#if ENABLE_HIP
+        hipError_t err = hipHostFree(data->pSrcGenericDesc);
```

Comment:
check nullptr before deleting, or ensure the hipHostFree handles nullptr

Reply:
hipHostFree handles nullptr